akka-actor-typed-tests/src/test/scala/akka/actor/typed/scaladsl/MessageAdapterSpec.scala

```diff
@@ -46,7 +46,7 @@ class MessageAdapterSpec
     deadLetterProbe.fishForMessage(deadLetterProbe.remainingOrDefault, s"looking for DeadLetter $expectedMessage") {
       deadLetter =>
         deadLetter.message match {
-          case AdaptMessage(msg, _) =>
+          case AdaptMessage(msg, _) if msg.getClass == expectedMessage.getClass =>
             msg shouldBe expectedMessage
```

Comment:
This is quite surprising. Each test case is using its own "inner" `case class Pong` and I would expect that different classes of `Pong` wouldn't be `equal`. Maybe something because they are "inner" classes like that, or something with the ScalaTest structure?

Anyway, I could easily reproduce the problem and this fixes it.

It's also not because both are named `Pong`. I tried to rename one of them.

Reply:
😮